contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+import {IPCVGuardian} from "./IPCVGuardian.sol";
+import {IPCVDeposit} from "./IPCVDeposit.sol";
+import {CoreRefPauseableLib} from "../libs/CoreRefPauseableLib.sol";
+
+contract PCVGuardian is IPCVGuardian, CoreRef {
+    using CoreRefPauseableLib for address;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    EnumerableSet.AddressSet private whitelistAddresses;
+
+    address public immutable safeAddress;
+
+    constructor(
+        address _core,
+        address _safeAddress,
+        address[] memory _whitelistAddresses
+    ) CoreRef(_core) {
+        safeAddress = _safeAddress;
+
+        for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+            _setWhitelistAddress(_whitelistAddresses[i]);
+        }
+    }
+
+    // ---------- Read-Only API ----------
+
+    /// @notice returns true if the the provided address is a valid destination to withdraw funds to
```

Comment:
this comment is incorrect, it only checks if this is a valid address to withdraw funds from